solrsearch-plugins/src/main/java/co.cask.hydrator.plugin/common/SolrSearchSinkConfig.java

```diff
@@ -116,4 +92,95 @@ public String getIdField() {
     }
     return outputFieldMap;
   }
+
+  /**
+   * Returns the SolrClient for establishing the connection with Solr server, using the hostname and port provided by
+   * user.
+   *
+   * @return Solr client
+   */
+  public SolrClient getSolrConnection() {
+    String urlString;
+    SolrClient solrClient = null;
+    if (solrMode.equals(SINGLE_NODE_MODE)) {
+      urlString = "http://" + solrHost + "/solr/" + collectionName;
+      solrClient = new HttpSolrClient(urlString);
+    } else if (solrMode.equals(SOLR_CLOUD_MODE)) {
+      CloudSolrClient solrCloudClient = new CloudSolrClient(solrHost);
+      solrCloudClient.setDefaultCollection(collectionName);
+      solrClient = solrCloudClient;
+    }
+    return solrClient;
+  }
+
+  /**
+   * Validates whether the host entered for Single Node Solr instance is proper or not.
+   */
+  public void validateSolrConnectionString() {
+    if (solrMode.equals(SINGLE_NODE_MODE) && solrHost.contains(",")) {
+      throw new IllegalArgumentException(String.format("Multiple hosts '%s' found for Single Node Solr.",
+                                                       solrHost));
+    }
+  }
+
+  /**
+   * Verifies whether the Solr server is running or not. Also, validates whether the collection provided by user is
+   * registered with the Solr server or not.
+   */
+  public void verifySolrConfiguration() {
+    SolrClient client = getSolrConnection();
+    try {
+      client.ping();
+    } catch (Exception e) {
+      throw new IllegalArgumentException(
+        String.format("Server refused connection at '%s'. Please make sure that either the Solr/Zookeeper services " +
+                        "are properly running or the collection '%s' exists in the Solr Server.", solrHost,
+                      collectionName));
+    }
+  }
+
+  /**
+   * Validates whether the Id field provided by user is present in the input schema or not.
+   *
+   * @param inputSchema
+   */
+  public void validateIdField(Schema inputSchema) {
+    if (inputSchema != null && inputSchema.getField(idField) == null) {
+      throw new IllegalArgumentException(
```

Comment:
Id field was kept nullable, assuming the scenario where the sink could receive a nullable schema from the previous stage. Restricting Id field schema to be non-nullable, would force the user to always remove the nulls before passing it to the Solr search sink. Therefore, we were handling the null values in transform.
Please let me know, if we still need to change this and have non nullable schema for Id field.